core/src/peer_id.rs

```diff
@@ -88,6 +89,17 @@ impl PeerId {
         }
     }
 
+    /// Tries to extract a [`PeerId`] from the given [`Multiaddr`].
+    ///
+    /// In case the given [`Multiaddr`] ends with `/p2p/<peer-id>`, this function
+    /// will return the encapsulated [`PeerId`], otherwise it will return `None`.
+    pub fn from_multiaddr(address: &Multiaddr) -> Option<PeerId> {
```

Comment:
```suggestion
    pub fn try_from_multiaddr(address: &Multiaddr) -> Option<PeerId> {
```

I wonder whether this would be more in line with the Rust naming conventions. What do you think?

Reply:
Implemented!